File: sdkwork-backend-sdk-python/sdkwork_backend_sdk/models/plus_host_domain_vo.py
from dataclasses import dataclass
from typing import Optional, List, Dict, Any

@dataclass
class PlusHostDomainVO:
    """域名管理VO"""
    created_at: str = None
    updated_at: str = None
    id: int = None
    uuid: str = None
    domain_name: str = None
    tld: str = None
    registrar: str = None
    owner_id: int = None
    registered_at: str = None
    expired_at: str = None
    status: str = None
    dns_servers: DnsServers = None
    resolution_status: str = None
    auto_renew: bool = None
    remark: str = None
    dns_records: List[PlusDnsRecordVO] = None

File: sdkwork-backend-sdk-typescript/src/types/plus-host-domain-vo.ts
import type { DnsServers } from './dns-servers';
import type { PlusDnsRecordVO } from './plus-dns-record-vo';

/** 域名管理VO */
export interface PlusHostDomainVO {
  /** 创建时间 */
  createdAt?: string;
  /** 更新时间 */
  updatedAt?: string;
  /** 主键ID，由数据库自动生成 */
  id?: number;
  /** 通用唯一标识符UUID */
  uuid?: string;
  /** 域名（例如：example.com） */
  domainName?: string;
  /** 顶级域名（例如：com, cn, net） */
  tld?: string;
  /** 注册商 */
  registrar?: string;
  /** 所属用户ID */
  ownerId?: number;
  /** 注册时间 */
  registeredAt?: string;
  /** 过期时间 */
  expiredAt?: string;
  /** 域名状态（ACTIVE-正常, EXPIRED-过期, PENDING- pending, SUSPENDED- suspended 等） */
  status?: 'ACTIVE' | 'EXPIRED' | 'PENDING' | 'SUSPENDED' | 'DELETED';
  /** DNS服务器列表 */
  dnsServers?: DnsServers;
  /** 解析状态（NORMAL-正常 / ABnormal-异常） */
  resolutionStatus?: 'NORMAL' | 'ERROR' | 'PAUSED' | 'FAILED';
  /** 是否自动续费 */
  autoRenew?: boolean;
  /** 备注信息 */
  remark?: string;
  /** 关联的DNS解析记录列表 */
  dnsRecords?: PlusDnsRecordVO[];
}
